src/js/messaging/actions/search.js

```diff
@@ -1,7 +1,17 @@
 export const TOGGLE_ADVANCED_SEARCH = 'TOGGLE_ADVANCED_SEARCH';
+export const SET_ADVSEARCH_START_DATE = 'SET_ADVSEARCH_START_DATE';
+export const SET_ADVSEARCH_END_DATE = 'SET_ADVSEARCH_END_DATE';
 
 export function toggleAdvancedSearch() {
   return {
     type: TOGGLE_ADVANCED_SEARCH
   };
 }
+
+export function setDateRange(date, start = true) {
+  const mode = (start === true) ? SET_ADVSEARCH_START_DATE : SET_ADVSEARCH_END_DATE;
```

Comment:
Can just check the truthy value for `start`, as in `const mode = (start) ? ...`.